src/TestInfrastructure/Helpers/MefTestHelpers.cs

```diff
@@ -180,5 +180,27 @@ public static void Compose(Assembly exportingAssembly, params object[] importers
             using var container = new CompositionContainer(catalog);
             container.Compose(batch);
         }
+
+        /// <summary>
+        /// Check if type has creation policy non shared.
+        /// </summary>
+        public static bool IsMefComponentNonShared(Type type)
+        {
+            var customAttributes = type.GetCustomAttributes(typeof(PartCreationPolicyAttribute), true);
+            var partCreationPolicyAttribute = (PartCreationPolicyAttribute)customAttributes[0];
+
+            return partCreationPolicyAttribute.CreationPolicy == CreationPolicy.NonShared;
+        }
+
+        /// <summary>
+        /// Check if type has creation policy shared.
+        /// </summary>
+        public static bool IsMefComponentShared(Type type)
+        {
+            var customAttributes = type.GetCustomAttributes(typeof(PartCreationPolicyAttribute), true);
+            var partCreationPolicyAttribute = (PartCreationPolicyAttribute)customAttributes[0];
+
+            return partCreationPolicyAttribute.CreationPolicy == CreationPolicy.Shared;
+        }
```

Comment:
This method is a duplicate of the other, except for the enum value being compared against.
I'd move the code into a helper method.